Added a parameter to define the Timeout of GetCurrentFIFOPacket

xaviergriffon opened the pull request:
I added the possibility to define the timeout value because if insufficient, the "GetCurrentFIFOPacket" method cannot be used.
For example, I drive brushless motors and to reduce the audible frequency, I increased the frequency of timer 0 (TCCR0B) and the values ​​of micros() and millis() are multiplied by 64.

xaviergriffon commented:
I added the possibility to define the timeout value because if insufficient, the "GetCurrentFIFOPacket" method cannot be used.
For example, I drive brushless motors and to reduce the audible frequency, I increased the frequency of timer 0 (TCCR0B) and the values ​​of micros() and millis() are multiplied by 64.

ZHomeSlice commented:
Your request for this parameter for timeouts seems reasonable but I am concerned about one change.
Removing the following line may leave us with an infinite loop
`if ((micros() - BreakTimer) > (11000)) return 0;`

we should never take more than 11000 microseconds to get a packet.
If we for some unknown reason take longer than 11000 microseconds to get a packet, something is preventing us from getting a packet other than software we need to bail and try next time.

Do you see this a redundant? 
Have you tested for and confirmed that this should be removed?

Z

xaviergriffon commented:
I didn't remove this line but I moved it to the start of the loop.
Over my tests, I realized that I could have the package but enter in the timeout. It would be a shame not to return anything when the expected result has been obtained.
So I moved to do the timeout control only if we not still have the expected result.
We can postpone at the end but we must add a test of result before the timeout.

```
bool packageReceived = false;
do {
   ...
   packageReceived = fifoC != length;
   if (!packageReceived && (micros() - BreakTimer) > (getFIFOTimeout())) return 0;
} while(!packetReceived);
```

Does that sound better to you?

Xavier

ZHomeSlice commented:
I like it. I don't see a problem now. 
Thanks for the explosion. 
I recommend the addition.
Z
